src/components/sbb-datepicker/sbb-datepicker.tsx

```diff
@@ -253,6 +263,28 @@ export class SbbDatepicker implements ComponentInterface {
     event.target.value = match?.[0] ?? null;
   }
 
+  private _getValidValue(value: string): string {
+    if (!value) {
+      return '';
+    }
+
+    const match: RegExpMatchArray = value.match(FORMAT_DATE);
+
+    if (match && match?.index === 0) {
+      return this._formatValue(value);
+    }
+
+    if (!Number.isNaN(+value) && Number.isInteger(+value)) {
+      const d = new Date(+value);
+      return this._composeValueString(`${d.getDate()}`, `${d.getMonth() + 1}`, d.getFullYear());
+    } else if (this._dateAdapter.isValid(new Date(value))) {
+      const d = new Date(value);
+      return this._composeValueString(`${d.getDate()}`, `${d.getMonth() + 1}`, d.getFullYear());
+    }
```

Comment:
The difference between the two is just the _+_ in the `new Date()`, right? Maybe a method can be extracted?